front-end/src/app/shared/services/transaction.service.ts

```diff
@@ -1,18 +1,35 @@
+import { DatePipe } from '@angular/common';
 import { Injectable } from '@angular/core';
 import { Observable } from 'rxjs';
 import { map } from 'rxjs/operators';
 import { TableListService } from '../interfaces/table-list-service.interface';
-import { Transaction } from '../interfaces/transaction.interface';
+import { Transaction } from '../models/transaction.model';
 import { ListRestResponse } from '../models/rest-api.model';
-import { SchATransaction } from '../models/scha-transaction.model';
 import { ApiService } from './api.service';
-import { SchATransactionService } from './scha-transaction.service';
+import { SchATransaction } from '../models/scha-transaction.model';
+import { SchBTransaction } from '../models/schb-transaction.model';
+import { TransactionType } from '../interfaces/transaction-type.interface';
+
+/**
+ * Given the API endpoint, return the class of the relevent schedule.
+ * @param key URL of root API endpoint
+ * @returns Transaction subclass
+ */
+function getScheduleClass(apiEndpoint: string) {
+  switch (apiEndpoint) {
+    case '/sch-a-transactions':
```

Comment:
the new app follows this pattern:
`/transactions/` for general endpoints
`/transactions/schedule-a/` for schedule a
`/transactions/schedule-b/` for schedule b

Reply:
Ok, that works. Nice and clear.